lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -212,6 +224,24 @@ def chefignore
         @chefignore ||= Chefignore.new(File.basename(cookbook_path))
       end
 
+      def load_all_unignored_files
+        # List all directories that are not dotdirs, then list everything in
+        # those, *including* dotfiles and dotdirs.
+        # Finally, list all the _files_ at the root, including dotfiles, and
+        # merge those in.
+        Dir.glob(File.join(Chef::Util::PathHelper.escape_glob(cookbook_path), "*")).each do |dir|
+          dir = Chef::Util::PathHelper.cleanpath(dir)
+          next unless File.directory?(dir)
+
+          Dir.glob(File.join(Chef::Util::PathHelper.escape_glob(dir), "**/*"), File::FNM_DOTMATCH).each do |file|
+            name = Chef::Util::PathHelper.relative_path_from(@cookbook_path, file)
+            cookbook_settings[:all_unignored_files][name] = file
+          end
+        end
+        load_root_files
+        cookbook_settings[:all_unignored_files].merge!(cookbook_settings[:root_filenames])
+      end
```

Comment:
bit of a run-on sentence there probably `root_entries.reject!` which would also avoid the extra allocation...